src/ol/render/canvas/canvasreplay.js

```diff
@@ -1440,13 +1484,20 @@ ol.render.canvas.PolygonReplay.prototype.setFillStrokeStyle = function(fillStyle
     var fillStyleColor = fillStyle.getColor();
     state.fillStyle = ol.colorlike.asColorLike(fillStyleColor ?
         fillStyleColor : ol.render.canvas.defaultFillStyle);
+    if (!this.transparency && ol.color.isRgba(state.fillStyle)) {
```

Comment:
Since `ol.color.isRgba()` is only used to test if alpha is not `1`, it would be simpler just to have `ol.color.isOpaque()` or something (rather than first testing if it is RGBA and then creating an array and then testing the alpha value).